### DISCUSSION TITLE
How should i use v2.2.9's new feature partition_key for insert with go-SDK

### DISCUSSION THREAD

**Discussion by Hypoce** [milvus-io/milvus#25291] [Q&A and General discussion]
I can't find a example for insert using partition_key with go-SDK, How should i use v2.2.9's new feature partition_key for insert with go-SDK？
My go-SDK version is 2.2.4

**Comment by yhmo** [comment]
The document of go sdk is not updated, we will update it soon.

There is a new method `entity.Field.WithIsPartitionKey(bool)`, set a field to WithIsPartitionKey(true) when you call the CreateCollection() to create a collection. Then insert entities as usual, it will dispatch the entities to different partitions automatically. Assume your partition key field's name is "abc", then you call `search(expr="abc in[xxxx]")`, it can automatically detect which partition has this value and only search within the partition.

**Reply by Hypoce** [reply]
I'm have set a field to ` WithIsPartitionKey(true) `,and set partitionName to "" when i called `client.Insert`,but i got an err " not support manually specifying the partition names if partition key mode is used" .
 May be i make something wrong, in need a example how to insert entity with go-SDK. 
Thanks.

**Reply by xiaofan-luan** [reply]
you have to specify partition key when you create collection.
